extensions/amp-iframe/0.1/amp-iframe.js

```diff
@@ -378,18 +362,45 @@ export class AmpIframe extends AMP.BaseElement {
   /**
    * Updates the element's dimensions to accommodate the iframe's
    *    requested dimensions.
-   * @param {number|undefined} newWidth
-   * @param {number|undefined} newHeight
+   * @param {number|undefined} height
+   * @param {number|undefined} width
    * @private
    */
-  updateSize_(newHeight, newWidth) {
+  updateSize_(height, width) {
     if (!this.isResizable_) {
       user.warn(TAG_,
           'ignoring embed-size request because this iframe is not resizable',
           this.element);
       return;
     }
-    this.attemptChangeSize(newHeight, newWidth);
+
+    let newHeight, newWidth;
+    if (height !== undefined) {
+      newHeight = Math.max(
+        (this.element./*OK*/offsetHeight - this.iframe_./*OK*/offsetHeight)
+            + height,
+        height);
+    }
+    if (width !== undefined) {
+      newWidth = Math.max(
+        (this.element./*OK*/offsetWidth - this.iframe_./*OK*/offsetWidth)
+            + width,
+        width);
+    }
+
+    if (newHeight !== undefined || newWidth !== undefined) {
+      this.attemptChangeSize(newHeight, newWidth, () => {
+        this.iframe_.height = height;
```

Comment:
I added the check for undefined values.
As for whether to set `this.iframe_.height`, I tried to test it using manual test. However, I think my manual test is not behaving as I expected. Setting or not setting `this.iframe_.height` and `this.iframe_.width` seems to be the same. In my manual test, I initialize the amp-iframe to be 300x300, but it's responsive so it's true dimension is larger. When I click the resize button, I send a resize request with width=300 and height=200. Then the iframe shrinks to exactly 300x200. I assume the correct behavior is displaying responsively (retain the width:height ratio of 3:2 but take the 100% width) instead of setting the dimension to exactly 300x200. Am I wrong about the expected behavior about resizing?